packages/@cdktf/hcl2cdk/lib/expressions.ts

```diff
@@ -175,25 +175,46 @@ export function extractReferencesFromExpression(
   }, [] as Reference[]);
 }
 
-export function referenceToVariableName(ref: Reference): string {
+export function referenceToVariableName(scope: Scope, ref: Reference): string {
   const parts = ref.referencee.full.split(".");
   const resource = parts[0] === "data" ? `${parts[0]}.${parts[1]}` : parts[0];
   const name = parts[0] === "data" ? parts[2] : parts[1];
-  return variableName(resource, name);
+  return variableName(scope, resource, name);
 }
 
-export function variableName(resource: string, name: string): string {
-  const proposedName = camelCase(
-    ["var", "local", "module"].includes(resource)
-      ? name
-      : [resource, name].join("_")
-  );
+function validVarName(name: string) {
+  if (!Number.isNaN(parseInt(name[0], 10))) {
```

Comment:
I didn't think just a number was valid in hcl, but I could imagine invalid names coming up, i.e. reserved keywords.

Reply:
I am checking for reserved words in JS now. Rosetta will handle / check for reserved words in other languages afaik.